game/scripts/npc/abilities/siltbreaker/siltbreaker_boss_protection.txt

```diff
@@ -0,0 +1,113 @@
+"DOTAAbilities"
+{
+  //=================================================================================================================
+  // SiltBreaker Boss Protection
+  //=================================================================================================================
+  "siltbreaker_boss_protection"
+  {
+    "ID"                                                  "8443"
+    "BaseClass"                                           "ability_datadriven"
+    "AbilityBehavior"                                     "DOTA_ABILITY_BEHAVIOR_PASSIVE"
+    "AbilityType"                                         "DOTA_ABILITY_TYPE_ULTIMATE"
+    "AbilityTextureName"                                  "silencer_last_word"
+    "MaxLevel"                                            "1"
+    "RequiredLevel"                                       "1"
+    "LevelsBetweenUpgrades"                               "1"
+
+    "AbilityCooldown"                                     "10.0"
+
+    "AbilitySpecial"
+    {
+      "01"
+      {
+        "var_type"                                        "FIELD_FLOAT"
+        "duration"                                        "5.0"
+      }
+    }
+
+    "OnUpgrade"
+    {
+      "ApplyModifier"
+      {
+        "ModifierName"                                    "modifier_siltbreaker_boss_protection_trigger"
+        "Target"                                          "CASTER"
+      }
+    }
+
+    "OnSpellStart"
+    {
+      "ApplyModifier"
+      {
+        "ModifierName"                                    "modifier_siltbreaker_boss_protection"
+        "Target"                                          "CASTER"
+        "Duration"                                        "%duration"
+      }
+      "RemoveModifier"
+      {
+        "ModifierName"                                    "modifier_siltbreaker_boss_protection_trigger"
+        "Target"                                          "CASTER"
+      }
+    }
+
+    "Modifiers"
+    {
+      "modifier_siltbreaker_boss_protection_trigger"
+      {
+        "Passive"                                         "1"
+        "IsHidden"                                        "1"
+        "IsDebuff"                                        "0"
+        "IsPurgable"                                      "0"
+        "IsBuff"                                          "1"
+        "IsStunDebuff"                                    "0"
+
+        "OnStateChanged"
```

Comment:
Ew, DataDriven. :P
Just out of curiosity, did the Lua OnStateChanged not work, or did you just want to try DataDriven?

Reply:
I wanted to try. Also the documentation is wrong. OnStateChange was defined as an ability event but it only works as modifier event